internal/servers/service_rotate_roots.go

```diff
@@ -0,0 +1,24 @@
+package servers
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/nodeenrollment"
+	"github.com/hashicorp/nodeenrollment/rotation"
+)
+
+// RotateRoots is a domain service function that initiates a rotation of root certificates
+// via a call to the nodenenrollment RotateRootCertificates function
+// Accepts the nodeenrollment option, WithCertificateLifetime(time.Duration) to specify the lifetime
+// of the generated cert(s)
+func RotateRoots(ctx context.Context, workerAuthRepo *WorkerAuthRepositoryStorage, opt ...nodeenrollment.Option) error {
+	const op = "servers.RotateRoots"
+
+	_, err := rotation.RotateRootCertificates(ctx, workerAuthRepo, opt...)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
```

Comment:
perhaps a unit test for this failure mode?

Reply:
What's the best way to force a failure in the underlying function?

Reply:
could you just create WorkerAuthRepositoryStorage with a writer == `&db.Db{}`?

Reply:
Ooo thanks!